packages/remix-dev/vite/plugin.ts

```diff
@@ -1075,6 +1075,9 @@ export const remixVitePlugin: RemixVitePlugin = (remixUserConfig = {}) => {
               : undefined,
           },
           optimizeDeps: {
+            entries: Object.values(ctx.remixConfig.routes).map((route) =>
```

Comment:
Looking at the docs for this option:

> If build.rollupOptions.input is specified, Vite will crawl those entry points instead

I wonder if the root cause is that we're only applying the Rollup build options when the `build` command is running:

https://github.com/remix-run/remix/blob/8f38118e44298d609224c6074ae6519d385196f1/packages/remix-dev/vite/plugin.ts#L1126